File: app/src/main/java/com/vikho305/isaho220/outstanding/data/repositories/PostRepository.java
package com.vikho305.isaho220.outstanding.data.repositories;

import android.content.Context;

import com.rx2androidnetworking.Rx2AndroidNetworking;
import com.vikho305.isaho220.outstanding.data.Comment;
import com.vikho305.isaho220.outstanding.data.Post;
import com.vikho305.isaho220.outstanding.data.api.PostApi;
import com.vikho305.isaho220.outstanding.data.api.UserApi;

import org.json.JSONException;
import org.json.JSONObject;

import java.util.List;

import io.reactivex.Single;

public class PostRepository {

    private PostApi postApi;

    public PostRepository(Context context) {
        postApi = new PostApi(context);
    }

    public Single<Post> getPost(String postId) {
        return postApi.getPost(postId);
    }

    public Single<List<Post>> getPostsInRadius(double radius) {
        return postApi.getPostsInRadius(radius);
    }

    public Single<List<Comment>> getComments(String postId, int page) {
        return postApi.getComments(postId, page);
    }

    public Single<String> postPost(String title, String text, String media, String mediaType,
                                   double latitude, double longitude) {
        return postApi.postPost(title, text, media, mediaType, latitude, longitude);
    }

    public Single<String> postComment(String postId, String text) {
        return postApi.postComment(postId, text);
    }

    public Single<String> likePost(String postId) {
        return postApi.likePost(postId);
    }

    public Single<String> dislikePost(String postId) {
        return postApi.dislikePost(postId);
    }

    public Single<String> unlikePost(String postId) {
        return postApi.unlikePost(postId);
    }

    public Single<String> undislikePost(String postId) {
        return postApi.undislikePost(postId);
    }
}

File: app/src/main/java/com/vikho305/isaho220/outstanding/data/repositories/PostRepository.kt
package com.vikho305.isaho220.outstanding.data.repositories

import android.content.Context
import com.vikho305.isaho220.outstanding.data.Comment
import com.vikho305.isaho220.outstanding.data.Post
import com.vikho305.isaho220.outstanding.data.api.PostApi
import io.reactivex.Single

class PostRepository(context: Context) {
    private val postApi: PostApi = PostApi(context)

    fun getPost(postId: String): Single<Post> {
        return postApi.getPost(postId)
    }

    fun getPostsInRadius(radius: Double): Single<List<Post>> {
        return postApi.getPostsInRadius(radius)
    }

    fun getComments(postId: String, page: Int): Single<List<Comment>> {
        return postApi.getComments(postId, page)
    }

    fun postPost(title: String?, text: String?, media: String?, mediaType: String?,
                 latitude: Double, longitude: Double): Single<String> {
        return postApi.postPost(title, text, media, mediaType, latitude, longitude)
    }

    fun postComment(postId: String, text: String?): Single<String> {
        return postApi.postComment(postId, text)
    }

    fun likePost(postId: String): Single<String> {
        return postApi.likePost(postId)
    }

    fun dislikePost(postId: String): Single<String> {
        return postApi.dislikePost(postId)
    }

    fun unlikePost(postId: String): Single<String> {
        return postApi.unlikePost(postId)
    }

    fun undislikePost(postId: String): Single<String> {
        return postApi.undislikePost(postId)
    }

}